src/networkWrapper/wrappers/socketio.js

```diff
@@ -12,11 +22,75 @@ function SocketIO(host, port, ssl) {
    * @param {int} reconnectionDelay
    */
   this.connect = function (autoReconnect, reconnectionDelay) {
+    var self = this;
+
     this.socket = window.io((this.ssl ? 'https://' : 'http://') + this.host + ':' + this.port, {
       reconnection: autoReconnect,
       reconnectionDelay: reconnectionDelay,
       forceNew: true
     });
+
+    this.socket.on('connect', function() {
+      if (self.wasConnected) {
```

Comment:
Well, I see that you intend to use them differently but this is not what the doc says. `connect` is fired each time a connection is made, including socket.io reconnections.